Src/LinearSolvers/MLMG/AMReX_MLEBABecLap.cpp

```diff
@@ -1345,7 +1440,7 @@ MLEBABecLap::applyBC (int amrlev, int mglev, MultiFab& in, BCMode bc_mode, State
 
     int m_is_inhomog = bc_mode == BCMode::Inhomogeneous;
     int flagbc = m_is_inhomog;
-    m_is_eb_inhomog = s_mode == StateMode::Solution;
+    if (!phi_on_centroid) m_is_eb_inhomog = s_mode == StateMode::Solution;
```

Comment:
I see why you made changes to `m_is_eb_inhomog` in `setEBDirichlet`.   Have to think about it.  Seems wrong to me.  If the multigrid solver is in correct form, the original phi on EB should not be used, I think.

Reply:
If by "correct form" you mean "residual-correction form", then wouldn't s_mode == StateMode::Correction if that were true?

Reply:
Yes, I meant correction form (i.e., residual-correction form).  `StateMode` has only two possible values, `Solution` and `Correction`.  So `!= Solution` is equivalent to ` == Correction`.